core-primitives/sgx/temp-dir/src/lib.rs

```diff
@@ -0,0 +1,168 @@
+//! # temp-dir
+//!
+//! Copied from the original tempdir crate with one line difference and sgx-support.
```

Comment:
Oh, I have actually not noticed. However, the current one is actually more complex and harder to port to no-std, so I think we will stick to this approach.

Further, in case there are uncovered security aspects, we don't have to worry about it because I don't see any reason to use it outside test & dev setups, but I shall document that!